sylvia/src/multitest.rs

```diff
@@ -39,22 +103,24 @@ impl App {
 }
 
 #[must_use]
-pub struct ExecProxy<'a, 'app, Error, Msg>
+pub struct ExecProxy<'a, 'app, Error, Msg, ExecC = Empty>
 where
-    Msg: Serialize + std::fmt::Debug,
-    Error: std::fmt::Debug + std::fmt::Display + Send + Sync + 'static,
+    Msg: Serialize + Debug,
+    Error: Debug + Display + Send + Sync + 'static,
 {
     funds: &'a [Coin],
     contract_addr: &'a Addr,
     msg: Msg,
     app: &'app App,
-    phantom: PhantomData<Error>,
+    phantom: PhantomData<(Error, ExecC)>,
 }
 
-impl<'a, 'app, Error, Msg> ExecProxy<'a, 'app, Error, Msg>
+impl<'a, 'app, Error, Msg, ExecC> ExecProxy<'a, 'app, Error, Msg, ExecC>
 where
-    Msg: Serialize + std::fmt::Debug,
-    Error: std::fmt::Debug + std::fmt::Display + Send + Sync + 'static,
+    Msg: Serialize + Debug,
+    Error: Debug + Display + Send + Sync + 'static,
+    ExecC: Debug + Clone + JsonSchema + PartialEq + 'static,
+    cw_multi_test::App: Executor<ExecC>,
```

Comment:
And this trait bound is always false for `ExecC` not being `Empty`, as `cw_multi_test::App` is a well-defined type monomorphisation. What you need here is to make your `app` a generic type (let's call it `MtApp` to not confuse with local `App`), and then here have trait bound:
```rust
MtApp: Executor<ExecC>,
```
This single line also tells me that you didn't write a single test calling those functions with custom `ExecC`.